vars/containerExecuteInsidePod.groovy

```diff
@@ -0,0 +1,92 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.SysEnv
+import groovy.transform.Field
+
+@Field def STEP_NAME = 'containerExecuteInsidePod'
+
+@Field Set GENERAL_CONFIG_KEYS = ['jenkinsKubernetes']
+
+@Field Set PARAMETER_KEYS = ['dockerOptions',
+                             'dockerWorkspace',
+                             'dockerEnvVars',
+                             'containersMap']
+
+@Field Set STEP_CONFIG_KEYS = PARAMETER_KEYS
+
+void call(Map parameters = [:], body) {
+    def uniqueId = UUID.randomUUID().toString()
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .withMandatoryProperty('containersMap')
+            .use()
+
+        def options = [name      : 'dynamic-agent-' + uniqueId,
+                       label     : uniqueId,
+                       containers: getContainerList(config)]
+        podTemplate(options) {
+            node(uniqueId) {
+                body()
+            }
+        }
+    }
+}
+
+private getContainerList(config) {
+    def envVars
+    envVars = getContainerEnvs(config)
+    result = []
+    result.push(containerTemplate(name: 'jnlp',
+        image: config.jenkinsKubernetes.jnlpAgent,
+        args: '${computer.jnlpmac} ${computer.name}'))
```

Comment:
is `computer` a global provided by the jenkins?

Reply:
ideally yes. But `args` are optional, so let me try running pipeline without them.